pkg/connector/connector.go

```diff
@@ -0,0 +1,86 @@
+// mautrix-twitter - A Matrix-Twitter puppeting bridge.
+// Copyright (C) 2024 Tulir Asokan
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+package connector
+
+import (
+	"context"
+	"fmt"
+
+	"maunium.net/go/mautrix/bridgev2"
+	"maunium.net/go/mautrix/bridgev2/database"
+)
+
+type TwitterConnector struct {
+	br *bridgev2.Bridge
+
+	Config *Config
+}
+
+var _ bridgev2.NetworkConnector = (*TwitterConnector)(nil)
+
+func NewConnector() *TwitterConnector {
+	return &TwitterConnector{}
+}
+
+func (tc *TwitterConnector) Init(bridge *bridgev2.Bridge) {
+	tc.br = bridge
+}
+
+func (tc *TwitterConnector) Start(_ context.Context) error {
+	return nil
+}
+
+func (tc *TwitterConnector) GetName() bridgev2.BridgeName {
+	return bridgev2.BridgeName{
+		DisplayName:      "Twitter",
+		NetworkURL:       "https://twitter.com",
+		NetworkIcon:      "mxc://maunium.net/HVHcnusJkQcpVcsVGZRELLCn",
+		NetworkID:        "twitter",
+		BeeperBridgeType: "twitter",
+		DefaultPort:      29327,
+	}
+}
+
+func (tc *TwitterConnector) GetDBMetaTypes() database.MetaTypes {
+	return database.MetaTypes{
+		Reaction: nil,
+		Portal:   nil,
+		Message:  nil,
+		Ghost:    nil,
+		UserLogin: func() any {
+			return &UserLoginMetadata{}
+		},
+	}
+}
+
+func (tc *TwitterConnector) GetCapabilities() *bridgev2.NetworkGeneralCapabilities {
+	return &bridgev2.NetworkGeneralCapabilities{}
+}
+
+func (tc *TwitterConnector) LoadUserLogin(ctx context.Context, login *bridgev2.UserLogin) error {
+	twitClient := NewTwitterClient(ctx, tc, login)
+
+	_, currentUser, err := twitClient.client.LoadMessagesPage()
+	if err != nil {
+		return fmt.Errorf("failed to load messages page")
+	}
```

Comment:
Loading data in `LoadUserLogin` is not allowed, connecting is done in `Connect()`

Reply:
as with another comment i left, it might have to do with the validation tokens etc. 

I think creating a twittermeow client automatically starts the polling - the approach in this case would be to stop polling until Connect() - noted!

Reply:
Actually, that's not an issue here, reference the client_test.go file.
https://github.com/mautrix/twitter/blob/d0a91623bbeabaa6ce1b5e70f921dc8233f4dbbc/pkg/twittermeow/client_test.go#L40-L45

This is a prerequisite to connecting it seems, could you confirm if even that is not allowed in this case?

Reply:
LoadUserLogin isn't allowed to do any network requests at all